### ISSUE TITLE
write_tiff crashes with complex data and geotag

### ISSUE THREAD

**Issue by brandy125** [gnudatalanguage/gdl#1911]
I am getting this error when trying to write a geotiff with complex data and geotag:

```
GDL> a=complex(dist(500),dist(500))
GDL> geo={modelpixelscaletag:[0,0,0]}
GDL> write_tiff,'test.tif',a,/complex,geo=geo
% UPDATEGEOTAGSINIMAGE: test.tif is not existing or is not a TIFF file!
% Error occurred at: UPDATEGEOTAGSINIMAGE   339 /usr/local/share/gnudatalanguage/lib/utilities/updategeotagsinimage.pro
%                    WRITE_TIFF         467 /usr/local/share/gnudatalanguage/lib/write_tiff.pro
%                    $MAIN$          
% Execution halted at: $MAIN$          
GDL> 
```

**Comment by brandy125** [comment]
Thanks @GillesDuvert for the patch.

Unfortunately I am not able to read the file back:
```
GDL> a=complex(dist(500),dist(500))
GDL> geo={modelpixelscaletag:[0,0,0]}
GDL> write_tiff,'test.tif',a,/complex,geo=geo
GDL> b=read_tiff('test.tif',geo=geon)
Unsupported PIXEL_TYPE: 0
% TIFF_READ: Failed to read TIFF image with defined parameters
% Error occurred at: READ_TIFF           27 /usr/local/share/gnudatalanguage/lib/read_tiff.pro
%                    $MAIN$          
% Execution halted at: $MAIN$          
```

**Comment by GillesDuvert** [comment]
hm I'm afraid `libtiff` does not read complex values. And we basically use `libtiff` FOR READING (witing, I did that directly in a GDL procedure) . I cannot figure if the current libtiff does or not read complex values. The documentation says nothing. Adding @rexso  for comment and action if possible.

**Comment by GillesDuvert** [comment]
actually it was not programmed by @rexso . Needed just a couple of lines. will be closed by #1938
